MobileCenter/MobileCenter/Internals/Utils/MSUtility+Application.m

```diff
@@ -30,31 +30,35 @@ + (UIApplication *)sharedApp {
 
 + (void)sharedAppOpenUrl:(NSURL *)url
                  options:(NSDictionary<NSString *, id> *)options
-       completionHandler:(void (^__nullable)(MSOpenURLState state))completion {
+       completionHandler:(void (^)(MSOpenURLState state))completion {
   UIApplication *sharedApp = [[self class] sharedApp];
 
   // FIXME: App extensions does support openURL through NSExtensionContest, we may use this somehow.
   if (MS_IS_APP_EXTENSION || ![sharedApp canOpenURL:url]) {
-    completion(MSOpenURLStateFailed);
+    if (completion) {
+      completion(MSOpenURLStateFailed);
+    }
     return;
   }
 
   // Dispatch the open url call to the next loop to avoid freezing the App new instance start up.
   dispatch_async(dispatch_get_main_queue(), ^{
     SEL selector = NSSelectorFromString(@"openURL:options:completionHandler:");
     if ([sharedApp respondsToSelector:selector]) {
-      id handler = ^(BOOL success) {
-        completion(success ? MSOpenURLStateSucceed : MSOpenURLStateUnknown);
+      id _url = url;
```

Comment:
underscore are for class private ivars to me you may want to remove it and rename.